src/core/reducers/api.js

```diff
@@ -7,8 +8,44 @@ import {
   SET_CLIENT_APP,
   SET_USER_AGENT,
 } from 'core/constants';
+import type {
+  SetAuthTokenAction,
+  LogOutUserAction,
+  SetClientAppAction,
+  SetLangAction,
+  SetUserAgentAction,
+} from 'core/actions/index';
+import type { Exact } from 'core/types/coreTypes';
 
-export default function api(state = {}, action) {
+type UserAgentInfoType = {|
+  browser: string,
+  os: string,
+|};
+
+export type ApiStateType = {
+  token: ?string,
+  lang: ?string,
+  clientApp: ?string,
+  userAgent: ?string,
+  userAgentInfo: ?UserAgentInfoType,
+};
+
+export const initialApiState = {
+  token: null,
+  lang: null,
+  clientApp: null,
+  userAgent: null,
+  userAgentInfo: null,
+};
+
+export default function api(
```

Comment:
Mostly I was confused by the `&` here: https://github.com/mozilla/addons-frontend/pull/2196/files/732cebdb6a9b2987b505687be588a4bfc94667ee#diff-503196c93f1c7e904f62dff0de0bb124R44